vignettes/miaViz.Rmd

```diff
@@ -326,6 +326,100 @@ names(metadata(dmn_se))
 plotDMNFit(dmn_se, type = "laplace")
 ```
 
+# Serial data ordination and trajectories
+
+```{r, error=FALSE, warning=FALSE, results='hide'}
+if (!requireNamespace("devtools", quietly = TRUE))
+    install.packages("devtools")
+devtools::install_github("microbiome/miaTime")
+```
+
+Principal Coordinates Analysis using Bray-Curtis dissimilarity on the
+`hitchip1006` dataset:
+
+```{r}
+library(dplyr)
+data(hitchip1006, package = "miaTime")
+tse <- hitchip1006
+tse <- transformCounts(tse, method = "relabundance")
+## Ordination with PCoA with Bray-Curtis dissimilarity
+tse <- runMDS(tse, FUN = vegan::vegdist, method = "bray",
+                  name = "PCoA_BC", exprs_values = "relabundance",
+              na.rm = TRUE)
+# Calculating Explained variance
+e <- attr(reducedDim(tse, "PCoA_BC"), "eig")
+rel_eig <- e/sum(e[e>0])
+# plot
+p <- plotReducedDim(tse, dimred = "PCoA_BC")
+
+p + labs(x = paste("PCoA 1 (", round(100 * rel_eig[[1]],1), "%", ")", sep = ""),
+         y = paste("PCoA 2 (", round(100 * rel_eig[[2]],1), "%", ")", sep = ""),
+         title="PCoA - Bray-Curtis")+
+    theme(plot.title = element_text(hjust = 0.5))
+```
+
+Retrieving information about all available trajectories:
+
+```{r}
+# List subjects with two time points
+selected.subjects <- names(which(table(tse$subject)==2))
+
+# Subjects counts per number of time points available in the data
+table(table(tse$subject)) %>% as.data.frame() %>%
+    rename(Timepoints=Var1, Subjects=Freq)
+```
+
+Lets look at all trajectories having two time points in the data:
+
+```{r}
+# adding metadata for the plot
+p$data <- cbind(p$data, colData(tse))
+# plot
+p + geom_path(aes(x=X, y=Y, group=subject), 
+              arrow=arrow(length = unit(0.1, "inches")),
+              data = subset(p$data, subject %in% selected.subjects))+
+    labs(title = "All trajectories with two time points")+
+    theme(plot.title = element_text(hjust = 0.5))
```

Comment:
I thought we converged into using the oather solution with

`data = subset(data.frame(reducedDim(tse), colData(tse)), subject %in% selected.subjects))`

That is perhaps even more clear. Can we still change?

Reply:
Is there a direct link to those comments, or could you paste here?

Reply:
first --> https://github.com/microbiome/miaViz/pull/65#discussion_r1057664461
second --> https://github.com/microbiome/miaViz/pull/65#discussion_r1057666385

Reply:
Yess - I have already replied to those it seems.